### ISSUE TITLE
keeping the plurality consistent. Right now we have things like /type (singular) and /amenities (plural) that both return the full list of types or amenities, respectively. This is just bad form no matter what. It could be singular or plural, but mixing both is not good. For reference, I tend to prefer the plural form, but no enough to debate it if someone else has a preference.

### ISSUE THREAD

**Issue by ZaneBartlett1** [DartExplore/dallas-by-metro-backend#222]

**Comment by Nightwing52** [comment]
The standard in DB design at least is to prefer singular.

**Comment by ZaneBartlett1** [comment]
Yeah I've thought about that. I honestly don't really like that in DB design, but it is a fair argument. I tend to prefer plural as it tends to make more sense when you think about getting back multiple items from a query.

**Comment by Nightwing52** [comment]
I think in method names I dont have a preference, but for routes I do have a strong preference for singular. Mostly due to convention. If we change anything a separate issue needs to be made on the frontend side.

**Comment by ZaneBartlett1** [comment]
For reference, this is what I read - https://stackoverflow.com/questions/6845772/should-i-use-singular-or-plural-name-convention-for-rest-resources on top of doing a bit of thinking for it. The database being singular is mentioned in one of the comments on the most rated threads. If you'd prefer it that way we can do it, but they mention that because it's auto generated, which we are not doing. I do understand why that would make sense though, just have the slight preference for the plural.

**Comment by ZaneBartlett1** [comment]
> I think in method names I dont have a preference, but for routes I do have a strong preference for singular. Mostly due to convention. If we change anything a separate issue needs to be made on the frontend side.

So this is actually where bindings on the front end become very useful. We actually have bindings implemented now on the new branch I'm working on with the planner, and I can update your code to use them. The reason they are so useful is because they abstract calling the API directly. Instead of doing that, you have a file on the frontend called client that has methods in it to call the API. So instead of doing `api.dallasbymetro.com/all` you would do something like `client.all` and the client file would have a method `all` in it that would call the API. This is nice for exactly this reason. If we choose to update `all` to something like `stationWithPOIs`, we only have to update that in the client file to fix it, as opposed to every possible instance of that API being called directly in the frontend.

**Comment by Nightwing52** [comment]
For sure, you'll only have to update the binding. Should be small enough to do it on your branch.
